packages/toast/src/ToastContainer/ToastContainer.styles.ts

```diff
@@ -0,0 +1,187 @@
+import { TransitionStatus } from 'react-transition-group';
+import { mix } from 'polished';
+
+import { css, cx } from '@leafygreen-ui/emotion';
+import { Theme } from '@leafygreen-ui/lib';
+import { palette } from '@leafygreen-ui/palette';
+import { spacing, transitionDuration } from '@leafygreen-ui/tokens';
+
+import { TOAST } from '../constants';
+import { toastBGColor } from '../InternalToast';
+
+import { calcTotalStackHeight } from './utils/useToastHeights';
+
+export const toastContainerStyles = css`
+  outline: 1px solid teal;
+  position: fixed;
+  left: ${spacing[3] - TOAST.inset}px;
+  bottom: ${spacing[3] - TOAST.inset}px;
+  width: ${TOAST.width + 2 * TOAST.inset}px;
+  max-height: calc(100vh - ${spacing[3]}px);
+  z-index: 0;
+  overflow: unset;
+
+  // Hide the toast initially
+  min-height: ${0}px;
+  opacity: 0;
+  visibility: hidden;
+
+  perspective: 1600px;
+  perspective-origin: bottom;
+  transform-style: preserve-3d;
+  transition: ease-in-out ${transitionDuration.default}ms;
+  transition-property: transform, bottom, height, opacity;
+
+  /* Scrollbars */
+  scroll-behavior: unset; // _not_ smooth. We need this to be instant
+  scrollbar-width: none; /* Firefox */
+  -ms-overflow-style: none; /* IE and old Edge */
+  &::-webkit-scrollbar {
+    display: none; /* Chrome, Safari and Opera */
+  }
+`;
+
+export const toastContainerVisibleStyles = css`
+  min-height: ${TOAST.minHeight + TOAST.yOffset}px;
+  opacity: 1;
+  visibility: visible;
+`;
+
+export function getContainerStatefulStyles({
+  recentToastsLength,
+  isHovered,
+  isExpanded,
+  shouldExpand,
+  totalStackHeight,
+  topToastHeight,
+  bottomOffset,
+}: {
+  recentToastsLength: number;
+  isHovered: boolean;
+  isExpanded: boolean;
+  shouldExpand: boolean;
+  totalStackHeight: number;
+  topToastHeight: number;
+  bottomOffset: number;
+}) {
+  const isInteracted = isHovered || isExpanded;
+
+  return cx(
+    css`
+      // The height of the first toast + inset
+      height: ${TOAST.inset * 2 + topToastHeight}px;
+
+      // Move the entire container as toasts get added,
+      // so the bottom toast is always 16px from the bottom
+      transform: translateY(-${TOAST.yOffset * (recentToastsLength - 1)}px);
+    `,
+    {
+      [css`
+        // set the container back when hovered/expanded
+        transform: translateY(0);
+        height: ${TOAST.inset * 2 + bottomOffset + totalStackHeight}px;
+      `]: isInteracted,
+
+      [css`
+        // We want auto scroll bars when expanded
+        height: 100vh;
+        overflow: auto;
+      `]: shouldExpand,
+    },
+  );
+}
```

Comment:
nit: it feels like this `cx` call should live on the component itself

Reply:
Generally feels like there are few functions in this file that could/should be `cx` calls instead of functions, or functions that are called in `cx` calls inside the component file. Similar to the cx call in [this](https://github.com/mongodb/leafygreen-ui/pull/1668/files#diff-0dfe10c35fa37d766d24e5444f598c29c9ed027ed2821cc557598175064f3205) component

Reply:
I'll move this last one (`shouldExpand`) to the component itself, but I felt moving it all to the component might cause a lot of clutter